src/MiniValidation/IValidatable.cs

```diff
@@ -0,0 +1,20 @@
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Threading.Tasks;
+
+namespace MiniValidation;
+
+/// <summary>
+/// Provides a way to add a validator for a type outside the class.
+/// </summary>
+/// <typeparam name="T"></typeparam>
+public interface IValidatable<in T>
+{
+    /// <summary>
+    /// Determines whether the specified object is valid.
+    /// </summary>
+    /// <param name="instance">The object instance to validate.</param>
+    /// <param name="validationContext">The validation context.</param>
+    /// <returns>A collection that holds failed-validation information.</returns>
+    Task<IEnumerable<ValidationResult>> ValidateAsync(T instance, ValidationContext validationContext);
```

Comment:
Should we have a non-async version? Or do we think given this is a net-new addition it's fine to just make it async-only? Do the non-async validate methods throw if an `IValidatable` is registered on the type you're trying to validate?

Reply:
It feels like this whole library could be simplified if it was async / valuetask only, but that would obviously be a breaking change and dataannotations doesn't support async either. I really wish .NET would modernize data annotations and make them more powerful. :-)

Anyway, it would be pretty trivial to add non-async support. Up to you.

Reply:
I think for net-new stuff it's fine to say it's async only. This needs to add logic to throw if the non-async validate method is called on a type that has one of these registered.